File: Motorcycle.java
public class Motorcycle extends Vehicle {
    private String engineSize;

    public Motorcycle(String id, String name, String brand, String model, String engineSize) {
        super(id, name, brand, model);
        this.engineSize = engineSize;
    }

    public String getEngineSize() {
        return engineSize;
    }
}

File: Motorcycle.js
const Vehicle = require('./Vehicle');

class Motorcycle extends Vehicle {
    constructor(id, name, brand, model, year, engineSize) {
        super(id, name, brand, model, year);
        this.engineSize = engineSize;
    }

    getStatusMessage() {
        return this.isReserved() ? 'This motorcycle is reserved' : 'This motorcycle is free';
    }
}

module.exports = Motorcycle; 
